doc/paper/paper.tex

```diff
@@ -0,0 +1,1012 @@
+    The largest outstanding design question relates to dp3’s choice to use MCAP
+    as the data format in leaf storage. There are a few benefits to this
+    choice: (1) No transcoding required on ingestion or query. (2)
+    Automatically compatible with user’s own internal tooling. (3) Streaming
+    playback is a row-oriented workload. (4) Automatically compatible with
+    common robotics recording workflows
+
+    There are also drawbacks, primarily in the performance department. (1)
+    Columnar storage would realize better compression in general. (2) A
+    standard columnar format may allow us to leverage an off-the-shelf query
+    executor, and could probably support more performant queries for our search
+    workload. (3) Building generic systems on MCAP requires supporting query
+    operations on multiple different binary encoding formats in place, which is
+    tedious to implement and support, and may carry some performance risks.
+
+    The question is also complicated by workload-specific factors. If a
+    columnar format is used, it is likely that small \q{row groups} or whatever
+    equivalent concept exists will be required, to enable queries to merge
+    large numbers of tables at once without using huge amounts of memory. Using
+    small row groups will tend to work against the interests of good OLAP
+    performance on the data files, obviating some of the benefits we would be
+    hoping to get with the columnar format.
+
+    Furthermore, a significant portion of the heavy read volume the database will
+    serve for ML purposes, will be image and point-cloud data. In these kinds of
+    messages, there is effectively one column: a giant byte array. This means that
+    row-oriented and columnar files for these kinds of data are physically pretty
+    similar. So intuitively, it seems like the performance benefits we would hope
+    to see may accrue mostly to the robotics developer’s log history searches, and
+    perhaps less to the ML developer's heavy read workloads.
+
+    The best way to resolve all these questions seems like an implementation spike
+    and some benchmarks. Parquet is an obvious candidate for a columnar format to
+    test. The LanceDB project is another one that is focused on some of the more
+    exotic kinds of searches users might be interested in, such as image similarity
+    search. Once we have a good sense of how all the options perform, we can choose
+    the best option and have a strong rationale.
+
+    A second area where outstanding questions exist is whether the structure we
+    have, with the statistics we can attach, is really sufficient to accelerate the
+    kinds of queries users want to run to be fast enough for a good user
+    experience. To get a better feel for this we will need to incorporate more
+    statistics and then get some users to evaluate.
+
+    \subsubsection{Feature development}
+    Assorted major feature development priorities are listed below.
```

Comment:
Not necessary to mention these are assorted, can write that we intend to develop the below (instead of saying we must develop them)?